### DISCUSSION TITLE
ffmpeg fails conversion during photoprism import

### DISCUSSION THREAD

**Discussion by tmzhuang** [photoprism/photoprism#3917] [Q&A]
PLEASE PROCEED ONLY IF YOU ARE ABSOLUTELY SURE THAT THIS IS NOT A TECHNICAL SUPPORT INCIDENT AND/OR POSSIBLY A PROBLEM WITH SOME OTHER SOFTWARE YOU ARE USING. VISIT <https://www.photoprism.app/kb/getting-support> TO LEARN MORE ABOUT OUR SUPPORT OPTIONS. THANK YOU FOR YOUR CAREFUL CONSIDERATION!

---------------------------------------------------------------------------

#### 1. What is not working as documented?

Getting the following error during a photoprism import:
```
import: ffmpeg version n6.0 Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 13.2.1 (GCC) 20230801
  configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-amf --enable-avisynth --enable-cuda-llvm --enable-lto --enable-fontconfig --enable-gmp --enable-gnutls --enable-gpl --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libdav1d --enable-libdrm --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libiec61883 --enable-libjack --enable-libjxl --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librav1e --enable-librsvg --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpl --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxcb --enable-libxml2 --enable-libxvid --enable-libzimg --enable-nvdec --enable-nvenc --enable-opencl --enable-opengl --enable-shared --enable-version3 --enable-vulkan
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/media/photos/2023/11/20231109_180340_9E0B4A83.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: isommp42
    creation_time   : 2023-11-09T18:03:40.000000Z
    com.android.version: 14
  Duration: 00:01:25.00, start: 0.000000, bitrate: 1112 kb/s
  Stream #0:0[0x1](eng): Video: h264 (Main) (avc1 / 0x31637661), yuv420p(tv, bt709, progressive), 1920x1080, 1077 kb/s, 15 fps, 15 tbr, 90k tbn (default)
    Metadata:
      creation_time   : 2023-11-09T18:03:40.000000Z
      handler_name    : VideoHandle
      vendor_id       : [0][0][0][0]
  Stream #0:1[0x2](eng): Audio: aac (LC) (mp4a / 0x6134706D), 8000 Hz, mono, fltp, 31 kb/s (default)
    Metadata:
      creation_time   : 2023-11-09T18:03:40.000000Z
      handler_name    : SoundHandle
      vendor_id       : [0][0][0][0]
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> mjpeg (native))
Press [q] to stop, [?] for help
[swscaler @ 0x55f84454e840] deprecated pixel format used, make sure you did set range correctly
[swscaler @ 0x55f84455dac0] deprecated pixel format used, make sure you did set range correctly
[swscaler @ 0x55f844626e00] deprecated pixel format used, make sure you did set range correctly
[swscaler @ 0x55f84455d0c0] deprecated pixel format used, make sure you did set range correctly
Output #0, image2, to '/media/photoprism/storage/sidecar/2023/11/20231109_180340_9E0B4A83.mp4.jpg':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: isommp42
    com.android.version: 14
    encoder         : Lavf60.3.100
  Stream #0:0(eng): Video: mjpeg, yuvj420p(pc, bt709, progressive), 1920x1080, q=2-31, 200 kb/s, 15 fps, 15 tbn (default)
    Metadata:
      creation_time   : 2023-11-09T18:03:40.000000Z
      handler_name    : VideoHandle
      vendor_id       : [0][0][0][0]
      encoder         : Lavc60.3.100 mjpeg
    Side data:
      cpb: bitrate max/min/avg: 0/0/200000 buffer size: 0 vbv_delay: N/A
[image2 @ 0x55f842fc1cc0] The specified filename '/media/photoprism/storage/sidecar/2023/11/20231109_180340_9E0B4A83.mp4.jpg' does not contain an image sequence pattern or a pattern is invalid.
[image2 @ 0x55f842fc1cc0] Use a pattern such as %03d for an image sequence or use the -update option (with -frames:v 1 if needed) to write a single image.
[image2 @ 0x55f842fc1cc0] Could not open file : /media/photoprism/storage/sidecar/2023/11/20231109_180340_9E0B4A83.mp4.jpg
av_interleaved_write_frame(): Input/output error
[out#0/image2 @ 0x55f842fc17c0] Error muxing a packet
frame=    1 fps=0.0 q=8.8 Lsize=N/A time=00:00:00.00 bitrate=N/A speed=   0x
video:150kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Conversion failed! in 2023/11/20231109_180340_9E0B4A83.mp4 (convert to jpeg)
```

#### 2. How can we reproduce it?
I've uploaded the video in question here: 
[video.tar.gz](https://github.com/photoprism/photoprism/files/13328896/video.tar.gz)

1. Photoprism import the video
2. Observe logs

When reporting an import, indexing, or performance issue, please include the number and type of pictures in your library, as well as any configuration options you have changed, such as for thumbnail quality.

#### 3. What behavior do you expect?

Video to appear in my photoprism homepage.

#### 4. What could be the cause of your problem?

Always try to determine the cause of your problem using the checklists at <https://docs.photoprism.app/getting-started/troubleshooting/> before submitting a bug report.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

Please include sample files or screenshots that help to reproduce your problem. You can also email files or share a download link, see <https://www.photoprism.app/contact> for details.

Visit <https://docs.photoprism.app/getting-started/troubleshooting/browsers/> to learn how to diagnose frontend issues.

**Important: If it is an import, indexing or metadata issue, we require sample files and logs from you.** Otherwise, we will not be able to process your report. If it is an import problem specifically, please always provide us with an archive of the files before you imported them so we can reproduce the behavior.

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64, ARM64, ARMv7,...
`photoprism version 231012-63f708417-Linux-AMD64`

(b) Database Type & Version: MariaDB, MySQL, SQLite,...
`mariadb`

(c) Operating System Types & Versions: Linux, Windows, Android,...
`Linux jupiter 6.6.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 08 Nov 2023 16:05:38 +0000 x86_64 GNU/Linux`

(d) Browser Types & Versions: Firefox, Chrome, Safari on iPhone,...
`Mozilla Firefox 119.0.1`

(e) Ad Blockers, Browser Plugins, and/or Firewall Software?
None

You can find the version/build number of the app in *Settings* by scrolling to the bottom. Note that MySQL 8 support has been discontinued, see system requirements at <https://docs.photoprism.app/getting-started/#system-requirements>.

*Always provide database and operating system details if it is a backend, import, or indexing issue. Should it be a frontend issue, at a minimum we require you to provide web browser and operating system details. When reporting a performance problem, we ask that you provide us with complete information about your environment, as there may be more than one cause.*

#### 7. On what kind of device is PhotoPrism installed?

This is especially important if you are reporting a performance, import, or indexing issue. You can skip this if you're reporting a problem you found in our public demo, or if it's a completely unrelated issue, such as incorrect page layout.

(a) Device / Processor Type: Raspberry Pi 4, Intel Core i7-3770, AMD Ryzen 7 3800X,...
Ryzen 5600X

(b) Physical Memory & Swap Space in GB
32G

(c) Storage Type: HDD, SSD, RAID, USB, Network Storage,...
SSD

(d) Anything else that might be helpful to know?

*Always provide device, memory, and storage details if you have a backend, performance, import, or indexing issue.*

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?
Reverse proxy `nginx/1.25.3`

If yes, please specify type and version. You can skip this if you are reporting a completely unrelated issue.

*Always provide this information when you have a reliability, performance, or frontend problem, such as failed uploads, connection errors, broken thumbnails, or video playback issues.*

**Using NGINX?** Please also provide the configuration and/or consider asking the NGINX community for advice as we do not specialize in supporting their product. Docs can be found at <https://docs.photoprism.app/getting-started/proxies/nginx/>.
Don't think my nginx config is import here since it looks like a ffmpeg issue but can provide it if needed.

**Comment by graciousgrey** [comment]
Unfortunately, I could not reproduce this. When I import the video you provided, it is imported and can be played in PhotoPrism without any problems. So it doesn't seem to be a general problem, but rather something specific to your setup. Did you use our docker compose example for the installation?

**Comment by lastzero** [comment]
`231012` doesn't seem to be an official build:
- https://docs.photoprism.app/release-notes/

**Comment by tmzhuang** [comment]
Grabbed it from https://aur.archlinux.org/. I'm not running it in a docker container (running directly on Arch Linux).

I'll give the docker image a try sometime to see if I can reproduce it there. In the meanwhile, is there anyway to list the files that failed to convert other than digging through the logs? The originals were moved, but I don't see it in my photoprism homepage since the conversion failed.

**Comment by graciousgrey** [comment]
You might find more information about the affected files on /library/errors or library/hidden

**Comment by lastzero** [comment]
> Grabbed it from https://aur.archlinux.org/. I'm not running it in a docker container (running directly on Arch Linux).

In this case, the problem is probably related to the FFmpeg version you have installed and/or the libraries for the individual codecs you have installed. We would like to help, but unfortunately don't have the capacity to do so. However, our source code is public, so you can help us by e.g. running this command manually on your computer to see if it works (or to find out e.g. if changes are needed for your version of FFmpeg or if you need to install additional codecs):

https://github.com/photoprism/photoprism/blob/8b5d5477730f2c58bc4a7c016eda8060f695c6e8/internal/photoprism/convert_image_jpeg.go#L29-L33

**Comment by lastzero** [comment]
I'll convert this into a discussion until it's clear how to reproduce the issue and if it's something that needs to be resolved in our code. If so, we'll be happy to provide a fix!

**Comment by srett** [comment]
The first part of the errors seems to be a red herring. It's just a warning that outputting to image files without having a placeholder in the output name is pointless, as it would repeatedly overwrite the same file. In this case it's obviously OK since we only want one frame anyways.

The real error appears to be
> [image2 @ 0x55f842fc1cc0] Could not open file : /media/photoprism/storage/sidecar/2023/11/20231109_180340_9E0B4A83.mp4.jpg
av_interleaved_write_frame(): Input/output error

Which sounds like a problem on the FS level.

**Comment by tmzhuang** [comment]
> The first part of the errors seems to be a red herring. It's just a warning that outputting to image files without having a placeholder in the output name is pointless, as it would repeatedly overwrite the same file. In this case it's obviously OK since we only want one frame anyways.
> 
> The real error appears to be
> 
> > [image2 @ 0x55f842fc1cc0] Could not open file : /media/photoprism/storage/sidecar/2023/11/20231109_180340_9E0B4A83.mp4.jpg
> > av_interleaved_write_frame(): Input/output error
> 
> Which sounds like a problem on the FS level.

I think it's a permission error on my end. I changed the USER= value of the systemd service that runs photoprism import. Have not had problems recently after settings correct group and permissions to the folder that was causing the issue (sidecar/2023/11/). Sorry for the wild goose chase. Once December hits and sidecar/2023/12/ gets created without a hitch I know I'll be in the clear. Thanks for the help!
